File: 1/solution.java
// 思路：读入两个整数 a、b，输出 a + b。
// 复杂度：时间 O(1)，空间 O(1)。
import java.io.BufferedInputStream;
import java.io.IOException;

public class Main {
    private static final class FastScanner {
        private final BufferedInputStream in = new BufferedInputStream(System.in);
        private final byte[] buffer = new byte[1 << 16];
        private int ptr = 0, len = 0;

        private int read() throws IOException {
            if (ptr >= len) {
                len = in.read(buffer);
                ptr = 0;
                if (len <= 0) return -1;
            }
            return buffer[ptr++];
        }

        long nextLong() throws IOException {
            int c;
            do {
                c = read();
                if (c == -1) return Long.MIN_VALUE;
            } while (c <= ' ');
            int sign = 1;
            if (c == '-') {
                sign = -1;
                c = read();
            }
            long val = 0;
            while (c > ' ') {
                val = val * 10 + (c - '0');
                c = read();
            }
            return val * sign;
        }
    }

    public static void main(String[] args) throws Exception {
        FastScanner fs = new FastScanner();
        long a = fs.nextLong();
        if (a == Long.MIN_VALUE) return;
        long b = fs.nextLong();
        if (b == Long.MIN_VALUE) return;
        System.out.print(a + b);
    }
}

File: 1/solution.kt
// 思路：读入两个整数 a、b，输出 a + b。
// 复杂度：时间 O(1)，空间 O(1)。
import java.io.BufferedInputStream

private class FastScanner {
    private val input = BufferedInputStream(System.`in`)
    private val buffer = ByteArray(1 shl 16)
    private var len = 0
    private var ptr = 0

    private fun readByte(): Int {
        if (ptr >= len) {
            len = input.read(buffer)
            ptr = 0
            if (len <= 0) return -1
        }
        return buffer[ptr++].toInt()
    }

    fun nextLong(): Long? {
        var c = readByte()
        while (c <= 32) {
            if (c == -1) return null
            c = readByte()
        }
        var sign = 1
        if (c == '-'.code) {
            sign = -1
            c = readByte()
        }
        var valNum = 0L
        while (c > 32 && c != -1) {
            valNum = valNum * 10 + (c - '0'.code)
            c = readByte()
        }
        return valNum * sign
    }
}

fun main() {
    val fs = FastScanner()
    val a = fs.nextLong() ?: return
    val b = fs.nextLong() ?: return
    print(a + b)
}
